src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.dev.cinema.service.impl;
+
+import com.dev.cinema.dao.ShoppingCartDao;
+import com.dev.cinema.dao.TicketDao;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.MovieSession;
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.ShoppingCartService;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
+        shoppingCart.setTickets(List.of(ticket));
```

Comment:
here we should add one else ticket but not change the old value

Reply:
Agree. Fixed